CppCoreGuidelines.md

```diff
@@ -6539,6 +6540,40 @@ That tends to work better than "cleverness" for non-specialists.
 
 * Flag throwing `hash`es.
 
+### <a name="Rc-memset"></a>C.90: Rely on constructors and assignment operators, not `memset` and `memcpy`
+
+##### Reason
+
+The standard C++ mechanism to construct a an instance of a type is to call its constructor. As specified in [C.41: A constructor should create a fully initialized object](#Rc-complete). No additional initialization, such as by `memcpy`, should be required.
+A type will provide a copy constructor and/or copy assignment operator to appropriately make a copy of the class, preserving the type's invariants.  Using memcpy to copy a non-trivially copyable type can result in slicing, or data corruption. 
+
+##### Example, bad
+
+    struct base
+    {
+        virtual void update() = 0;
+    };
+
+    struct derived : public base
+    {
+        void update() override {}
+    };
+
+    // goodbye v-tables
+    void f(derived& a, derived& b)
+    {
+        memset(&a, 0, sizeof(derived));
+        memcpy(&a, &b, sizeof(derived));
+    }
+
+    // the authors of the class should ensure a proper assignment operator
+    // exists for the class
+    void g(derived& a, derived& b)
+    {
+        a = {};
```

Comment:
In addition to MikeGitb's comment, this also assumes ` = {};` is unambiguous. It might be better to say `a = derived{};` instead.

Reply:
Dropped.  LMK if you find the original example that Jordan provided compelling enough to rework and restore.